### ISSUE TITLE
Forbid to mix imports types

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1019]
# Rule request

## Thesis

When we write code with lot's of imports it is easy to forget to watch for already existing imports. So, we can end up with a situation like this:

```python
from a import b
from a.b import c
```

This is error-prone! Because you can write `b.c` and `c` which is the same thing!
Your code will start taking damage from it.

**Comment by uhbif19** [comment]
Do I understand correctly, that this is about forbidding only the imports leading to the same thing represented in two different ways? Is your import example the only one possible, or there can be more complex cases exist?

**Comment by sobolevn** [comment]
> Do I understand correctly, that this is about forbidding only the imports leading to the same thing represented in two different ways?

Yes

> Is your import example the only one possible, or there can be more complex cases exist?

I cannot think about any other cases. It might be worth to check cases like:

```python
from a import b
from a.b.c import d
```

But, I guess it is a different thing.

**Comment by kxepal** [comment]
With hypothesis common pattern is
```
import hypothesis
from hypothesis import strategies as st
```

While you can do `hypothesis.strategies` with a single import, it's too long to write all the time. Does this case violates proposed rule?

**Comment by sobolevn** [comment]
@kxepal that's a great case! aliases should be allowed.

**Comment by skarzi** [comment]
Can I prepare PR for this issue?

**Comment by sobolevn** [comment]
Yes, please!
